packages/contracts-bedrock/src/L2/CrossL2Inbox.sol

```diff
@@ -133,12 +139,31 @@ contract CrossL2Inbox is ISemver, TransientReentrancyAware {
     /// @param _id      Identifier of the message.
     /// @param _msgHash Hash of the message payload to call target with.
     function validateMessage(Identifier calldata _id, bytes32 _msgHash) external {
-        // We need to know if this is being called on a depositTx
-        if (IL1BlockInterop(Predeploys.L1_BLOCK_ATTRIBUTES).isDeposit()) revert NoExecutingDeposits();
+        // // We need to know if this is being called on a depositTx
+        // if (IL1BlockInterop(Predeploys.L1_BLOCK_ATTRIBUTES).isDeposit()) revert NoExecutingDeposits();
+
+        (bool _isSlotWarm,) = _isWarm(_hashSlot(_id, _msgHash));
+
+        if (!_isSlotWarm) revert NotWarm();
 
         emit ExecutingMessage(_msgHash, _id);
     }
 
+    function _hashSlot(Identifier calldata _id, bytes32 _msgHash) internal pure returns (bytes32 _slot) {
+        // TODO: ensure the way of keccaking is correct
+        _slot = keccak256(abi.encode(_id, _msgHash));
```

Comment:
It's not compatible with supervisor as-is. But here's a spec proposal for the formatting, to make it work well with the supervisor: https://github.com/ethereum-optimism/specs/pull/612